modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -338,6 +348,35 @@ object ConsensusStateUpdater {
     private def pickMajority[A: Order](proposals: List[A]): Option[A] =
       proposals.foldMap(a => Map(a -> 1)).toList.map(_.swap).maximumOption.map(_._2)
 
+    private def pickValidatedMajority(
+      lastSignedArtifact: Signed[Artifact],
+      trigger: ConsensusTrigger,
+      resources: ConsensusResources[Artifact]
+    )(proposals: List[Hash]): F[Option[Hash]] =
+      proposals
+        .foldMap(a => Map(a -> 1))
+        .toList
+        .map(_.swap)
+        .sortBy { case (occurrences, hash) => (-occurrences, hash) }
+        .foldLeftM(none[Hash]) {
+          case (None, (occurrences, majorityHash)) =>
+            resources.artifacts
+              .get(majorityHash)
+              .traverse { artifact =>
+                consensusFns.isValidProposal(lastSignedArtifact, trigger)(artifact).map(_.toOption)
+              }
+              .map(_.flatten)
+              .map(_.map(_ => majorityHash))
+              .flatTap {
+                _.fold {
+                  logger.warn(s"Found invalid majority hash=${majorityHash.show} with occurrences=${occurrences}, selecting next majority")
+                } { _ =>
+                  logger.debug(s"Found valid majority hash=${majorityHash.show} with occurrences=${occurrences}")
```

Comment:
Let's not add this log. Across all the codebase we log only anomalies (invalid things or errors), and we implicitly assume that the other 99,99% of things are valid.

Reply:
That's why I used `debug` flag that we use for all debugging things (not a warn or error). I can remove it but I wanted to be able to see occurrences when debugging for misalignments.

Reply:
Ok, but it doesn't add too much information. We already log the hash of the majority hash during the state transition.